### PR TITLE
fix(backend): fix unresolved auth plugin and invalid white routes bb-11

### PR DESCRIPTION
**Pull request by KeidsID**

### CODE REVIEW

**Comment by KeidsID** [comment]
🫠

File: apps/backend/src/libs/modules/server-application/base-server-application.ts [hunk 1]
```diff
@@ -214,6 +214,7 @@ class BaseServerApplication implements ServerApplication {
 					token,
 					userService,
 					whiteRoutes: WHITE_ROUTES,
+					whiteRoutesPrefix: `/${api.version}`,
```
Comment (fshabanov) [inline apps/backend/src/libs/modules/server-application/base-server-application.ts:217]: Add prefix directly in white routes constant

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by WilhelmWesser — APPROVED** [review]

File: apps/backend/src/libs/modules/server-application/base-server-application.ts (hunk 1 above)
Comment (KeidsID) [reply apps/backend/src/libs/modules/server-application/base-server-application.ts:217]: Something like this?
```ts
await this.app.register(authorizationPlugin, {
  token,
  userService,
  whiteRoutes: WHITE_ROUTES.map((whiteRoute) => `/${api.version}${whiteRoute}`),
});

```

Or directly on the constant value?

Comment (KeidsID) [reply apps/backend/src/libs/modules/server-application/base-server-application.ts:217]: Is there a constant for api version?
```ts
import { AuthApiPath } from "~/modules/auth/auth.js";

import { APIPath } from "../enums/enums.js";

const WHITE_ROUTES: string[] = [
	APIPath.AUTH,
	`${APIPath.AUTH}${AuthApiPath.SIGN_IN}`,
	`${APIPath.AUTH}${AuthApiPath.SIGN_UP}`,
].map((whiteRoute) => `/v1${whiteRoute}`);

export { WHITE_ROUTES };

```

Comment (fshabanov) [reply apps/backend/src/libs/modules/server-application/base-server-application.ts:217]: No, there's no constant. You can create a helper function, which will check if a route is a white route. For that you can use Regex - something like /api/v{number}/endpoint

File: apps/backend/src/libs/plugins/authorization-plugin/authorization-plugin.ts
Comment (fshabanov) [inline apps/backend/src/libs/plugins/authorization-plugin/authorization-plugin.ts:1]: Line 36 - you should take "Bearer " out of header, before decoding the token.

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**